antenna/ext/s3/connection.py

```diff
@@ -140,10 +161,25 @@ def check_health(self, state):
             # which is unhelpful for granularity purposes.
             ClientError,
         ],
+        wait_time_generator=wait_times_save,
         sleep_function=gevent.sleep,
         module_logger=logger,
     )
     def save_file(self, path, data):
+        """Saves a single file to s3
+
+        This will retry a handful of times in short succession so as to deal
+        with # some amount of fishiness. After that, the caller should retry
+        saving after a longer period of time.
+
+        :arg str path: the path to save to
+
+        :arg str data: the data to save
+
+        :raises botocore.exceptions.ClientError: connection issues, permissions
+            issues, bucket is missing, etc.
```

Comment:
This (and other docstring fixes in this PR) codifies who's in charge of dealing with what exceptions. That's important for understanding retryable things and their effects on other things.